YoutubeExplode/Internal/SegmentedHttpStream.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System;
+using System.IO;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace YoutubeExplode.Internal
+{
+    internal class SegmentedHttpStream : Stream
+    {
+        private readonly HttpClient _httpClient;
+
+        // Maximum segment size for rate limited streams
+        private const int MaxSegmentSize = 9_898_989; // this number was carefully devised through research
```

Comment:
I would prefer if this was not a constant but rather passed as a value to constructor for throttled streams

Reply:
👍